packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -679,6 +679,18 @@ export class SnippetString {
         return this;
     }
 
+    appendChoice(values: string[], number: number = this._tabstop++): SnippetString {
+        const value = values.map(s => s.replace(/\$|}|\\|,/g, '\\$&')).join(',');
+
+        this.value += '${';
+        this.value += number;
+        this.value += '|';
+        this.value += value;
+        this.value += '|}';
```

Comment:
I know this is copied directly from [vscode](https://github.com/microsoft/vscode/blob/3147fc7d8e6954fa3d8afc23c2b704903a3019b4/src/vs/workbench/api/common/extHostTypes.ts#L855-L865), but can't this be done in one line?


```ts
this.value += `\$\{${number}|${value}|\}`;
```

Reply:
You are right, I adapted it.